mio/stream_daq.py

```diff
@@ -143,38 +144,6 @@ def nbuffer_per_fm(self) -> int:
             self._nbuffer_per_fm = len(self.buffer_npix)
         return self._nbuffer_per_fm
 
-    def _parse_header(self, buffer: bytes) -> Tuple[StreamBufferHeader, np.ndarray]:
```

Comment:
i don't really have a strong opinion, i'm just not sure why this method had to die? fine with me tho

Reply:
Also related to the perf comment in the read loop, I wanted to isolate the numpy's vectorized operations from the pydantic because I had a bit of resistance in putting any pydantic in the read loop (not sure if it matters, but I recently had some trouble with the speed of pydantic stuff so just to be safe).

Pulling numpy operations out resulted in two lines, so this method died, but I guess this small method makes it easier to read. I'll move this back to something that grabs the header `ndarray` and returns the same before merging later.

Reply:
fine with it either way :) yes definitely agree should only ever use `model_construct` with pydantic models in perf-sensitive areas, validation is a luxury reserved for slower data

Reply:
ok the tiny _parse_header method is back